src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -12,16 +14,56 @@ namespace Microsoft.Extensions.Logging.Console
     [UnsupportedOSPlatform("browser")]
     internal class ConsoleLoggerProcessor : IDisposable
     {
-        private const int _maxQueuedMessages = 1024;
+        private readonly Queue<LogMessageEntry> _messageQueue;
+        private volatile int _messagesDropped;
+        private bool _isAddingCompleted;
+        private int _maxQueuedMessages = ConsoleLoggerOptions.DefaultMaxQueueLengthValue;
+        public int MaxQueueLength
+        {
+            get => _maxQueuedMessages;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} must be larger than zero.");
+                }
 
-        private readonly BlockingCollection<LogMessageEntry> _messageQueue = new BlockingCollection<LogMessageEntry>(_maxQueuedMessages);
+                lock (_messageQueue)
+                {
+                    _maxQueuedMessages = value;
+                    Monitor.PulseAll(_messageQueue);
+                }
+            }
+        }
+        private ConsoleLoggerBufferFullMode _fullMode = ConsoleLoggerBufferFullMode.Wait;
+        public ConsoleLoggerBufferFullMode FullMode
+        {
+            get => _fullMode;
+            set
+            {
+                if (value != ConsoleLoggerBufferFullMode.Wait && value != ConsoleLoggerBufferFullMode.DropWrite)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} is not a supported buffer mode value.");
+                }
+
+                lock (_messageQueue)
+                {
+                    // _fullMode is used inside the lock and is safer to guard setter with lock as well
+                    // this set is not expected to happen frequently
+                    _fullMode = value;
```

Comment:
Why don't we need a `Monitor.PulseAll(_messageQueue);` here?

Reply:
no thread would be waiting on full mode to change

Reply:
oh seems like this comment is related, https://github.com/dotnet/runtime/pull/70186#discussion_r901990232
will update